onnxruntime/python/tools/transformers/models/phi2/convert_to_onnx.py

```diff
@@ -0,0 +1,544 @@
+        assert function_name is not None
+        model = self.unroll_function(model, function_name)
+        model = self.update_edges(model, self.phi2_edge_dict)
+        model = self.simplify_phi2_op_type(model)
+        model = self.remove_dropout_layer(model)
+        onnx.save_model(
+            model,
+            onnx_path_out,
+            save_as_external_data=True,
+            all_tensors_to_one_file=True,
+            location=onnx_path_out + ".data",
+        )
+
+    def optimize_phi2_onnx(self, onnx_path: str, onnx_path_opt: str):
+        from fusion_options import FusionOptions
+        from optimizer import optimize_model
+        import uuid
+
+        processed_onnx_path = f"{uuid.uuid1()}.onnx"
+        model = onnx.load_model(onnx_path, load_external_data=True)
+        model = self.process_graph_io(self.phi_config, model)
+        onnx.save_model(
+            model,
+            processed_onnx_path,
+            save_as_external_data=True,
+            all_tensors_to_one_file=True,
+            location=processed_onnx_path + ".data",
+        )
+
+        optimization_options = FusionOptions("phi")
+        optimizer = optimize_model(
+            processed_onnx_path,
+            model_type="phi",
+            num_heads=self.phi_config.num_attention_heads,
+            hidden_size=self.phi_config.hidden_size,
+            opt_level=0,
+            optimization_options=optimization_options,
+            only_onnxruntime=False,
+        )
+
+        self.erase_onnx_model(processed_onnx_path)
+
+        if self.precision == Precision.FLOAT32:
+            optimizer.save_model_to_file(onnx_path_opt, use_external_data_format=True)
+            return
+
+        if (
+            self.precision == Precision.FLOAT16 or self.precision == Precision.INT4
+        ) and self.attn_op_type != AttentionOpType.MultiHeadAttention:
+            node_block_list = [
+                "GroupQueryAttention_29",
+                "GroupQueryAttention_30",
+                "GroupQueryAttention_31",
+                "Attention_29",
+                "Attention_30",
+                "Attention_31",
+            ]
+            logging.info(f"Converting onnx model to float16/bfloat16...")
+            optimizer.convert_float_to_float16(
+                keep_io_types=False,
+                node_block_list=node_block_list,
+                use_symbolic_shape_infer=True,
+                use_bfloat16_as_blocked_nodes_dtype=True
+                if self.attn_op_type == AttentionOpType.GroupQueryAttention
+                else False,
+            )
+
+        if self.precision == Precision.FLOAT16:
+            optimizer.save_model_to_file(onnx_path_opt, use_external_data_format=True)
+            return
+        else:
+            assert self.precision == Precision.INT4
+            quant = MatMul4BitsQuantizer(
+                model=optimizer.model,
+                block_size=16,
+                is_symmetric=True,
+            )
```

Comment:
would it be possible to make `block_size` and `accuracy_level` configurable?

as in the llama convert_to_onnx.py:
https://github.com/microsoft/onnxruntime/blob/c379a89bcb26bc4838efe70fae04760106e8d081/onnxruntime/python/tools/transformers/models/llama/convert_to_onnx.py#L978-L984

Reply:
thanks for adding it